### PR TITLE
add security context to helm hooks and jobs

### PR DESCRIPTION
**Pull request by pgvishnuram**
## Description

add security context to helm hooks and jobs for houston service

## Related Issues

https://github.com/astronomer/issues/issues/6363

## Testing

QA should able to see securityContext section in all cronjobs to validate

## Merging

cherry-pick to release-0.35 and relese-0.34

### CODE REVIEW

**Review by danielhoherd — COMMENTED** [review]
Seems fine so far.

File: tests/__init__.py [hunk 1]
```diff
@@ -35,3 +35,17 @@ def get_containers_by_name(doc, include_init_containers=False) -> dict:
         )
 
     return c_by_name
+
+
+def get_cronjob_containerspec_by_name(doc) -> dict:
+    """Given a single doc, return all the containers by name.
+
+    doc must be a valid spec for a pod manager. (EG: ds, sts)
```
Comment (danielhoherd) [inline tests/__init__.py:43]: Is this comment right? AFAIK only cronjobs have a jobTemplate, and the function name specifies cronjob.

```
$ git grep -l jobTemplate tests/k8s_schema/
tests/k8s_schema/v1.25.0-standalone/cronjob-batch-v1.json
tests/k8s_schema/v1.26.0-standalone/cronjob-batch-v1.json
tests/k8s_schema/v1.27.0-standalone/cronjob-batch-v1.json
tests/k8s_schema/v1.28.0-standalone/cronjob-batch-v1.json
tests/k8s_schema/v1.29.0-standalone/cronjob-batch-v1.json
tests/k8s_schema/v1.30.0-standalone/cronjob-batch-v1.json
```

```suggestion
    doc must be a valid spec for a CronJob.
```

File: tests/chart_tests/test_astronomer_houston_airflow_db_cleanup.py [hunk 2]
```diff
@@ -64,12 +69,18 @@ def test_astronomer_airflow_db_cleanup_cron_custom_schedule(self, kube_version):
...
 
         assert len(docs) == 1
+        job_container_by_name = get_cronjob_containerspec_by_name(docs[0])
         assert docs[0]["kind"] == "CronJob"
         assert (
             docs[0]["metadata"]["name"]
             == "release-name-houston-cleanup-airflow-db-data"
         )
         assert docs[0]["spec"]["schedule"] == "22 5 * * *"
+        print(job_container_by_name["cleanup"]["securityContext"])
```
Comment (danielhoherd) [inline tests/chart_tests/test_astronomer_houston_airflow_db_cleanup.py:79]:
```suggestion
```

File: tests/chart_tests/test_astronomer_houston_hook_job.py [hunk 3]
```diff
@@ -18,9 +18,58 @@ def test_au_strategy_job_defaults(self, kube_version):
...
+            kube_version=kube_version,
+            values={},
+            show_only=[
+                "charts/astronomer/templates/houston/helm-hooks/houston-upgrade-deployments-job.yaml"
+            ],
+        )
+        assert len(docs) == 1
+        c_by_name = get_containers_by_name(docs[0], include_init_containers=True)
+        assert docs[0]["kind"] == "Job"
+        assert docs[0]["metadata"]["name"] == "release-name-houston-upgrade-deployments"
```
Comment (danielhoherd) [inline tests/chart_tests/test_astronomer_houston_hook_job.py:45]:
```suggestion
        assert c_by_name["wait-for-db"]["kind"] == "Job"
        assert c_by_name["wait-for-db"]["metadata"]["name"] == "release-name-houston-upgrade-deployments"
```

File: tests/chart_tests/test_astronomer_houston_hook_job.py [hunk 4]
```diff
@@ -18,9 +18,58 @@ def test_au_strategy_job_defaults(self, kube_version):
...
+        assert c_by_name["post-upgrade-job"]["args"] == [
+            "yarn",
+            "upgrade-deployments",
+            "--",
+            "--canary=false",
+        ]
+
+        assert c_by_name["post-upgrade-job"]["securityContext"] == {
+            "runAsNonRoot": True
+        }
```
Comment (danielhoherd) [inline tests/chart_tests/test_astronomer_houston_hook_job.py:62]:
```suggestion
        assert c_by_name["wait-for-db"]["kind"] == "Job"
        assert c_by_name["wait-for-db"]["metadata"]["name"] == "release-name-houston-upgrade-deployments"
        assert c_by_name["wait-for-db"]["securityContext"] == {"runAsNonRoot": True}
        assert c_by_name["houston-bootstrapper"]["securityContext"] == {
            "runAsNonRoot": True
        }
        assert c_by_name["post-upgrade-job"]["args"] == [
            "yarn",
            "upgrade-deployments",
            "--",
            "--canary=false",
        ]
        assert c_by_name["post-upgrade-job"]["securityContext"] == {
            "runAsNonRoot": True
        }
```

File: tests/chart_tests/test_astronomer_houston_task_metrics.py [hunk 5]
```diff
@@ -30,19 +30,24 @@ def test_astronomer_cleanup_task_usage_cron_feature_enabled(self, kube_version):
 
         assert len(docs) == 1
         assert docs[0]["kind"] == "CronJob"
+        job_container_by_name = get_cronjob_containerspec_by_name(docs[0])
         assert (
             docs[0]["metadata"]["name"]
```
Comment (danielhoherd) [inline tests/chart_tests/test_astronomer_houston_task_metrics.py:35]:
```suggestion
            job_container_by_name["cleanup"]["metadata"]["name"]
```

File: tests/chart_tests/test_astronomer_houston_task_metrics.py [hunk 6]
```diff
@@ -30,19 +30,24 @@ def test_astronomer_cleanup_task_usage_cron_feature_enabled(self, kube_version):
...
             == "release-name-houston-cleanup-task-usage-data"
         )
         assert docs[0]["spec"]["schedule"] == "40 23 * * *"
```
Comment (danielhoherd) [inline tests/chart_tests/test_astronomer_houston_task_metrics.py:38]:
```suggestion
        assert job_container_by_name["cleanup"]["spec"]["schedule"] == "40 23 * * *"
```

File: tests/chart_tests/test_poddisruptionbudget.py [hunk 7]
```diff
@@ -11,7 +11,6 @@ def test_houston_pdb_cronjobs(self):
             "charts/astronomer/templates/houston/cronjobs/houston-expire-deployments-cronjob.yaml",
             "charts/astronomer/templates/houston/cronjobs/houston-check-updates-cronjob.yaml",
             "charts/astronomer/templates/houston/cronjobs/houston-cleanup-deployments-cronjob.yaml",
-            "charts/astronomer/templates/houston/cronjobs/houston-check-updates-cronjob.yaml",
```
Comment (danielhoherd) [inline tests/chart_tests/test_poddisruptionbudget.py:14]: Why was this removed?

**Review by danielhoherd — APPROVED** [review]
This looks great! Fantastic job on the tests, that is a huge confidence builder!

File: tests/chart_tests/test_poddisruptionbudget.py (hunk 7 above)
Comment (pgvishnuram) [reply tests/chart_tests/test_poddisruptionbudget.py:14]: its seems to be a typo file no 2 and fine no r seems identical

**Comment by danielhoherd** [comment]
Looks good. I've filed a follow-up ticket to remove the dockerd related bits of our helm charts. https://github.com/astronomer/issues/issues/6408
